TYPO3.Flow/Classes/TYPO3/Flow/Resource/ResourceManager.php

```diff
@@ -411,6 +413,39 @@ public function getPublicPackageResourceUri($packageKey, $relativePathAndFilenam
     }
 
     /**
+     * Returns the public URI for a static resource provided by the public package
+     *
+     * @param string $path
+     * @return string
+     * @throws Exception
+     * @api
+     */
+    public function getPublicPackageResourceUriByPath($path)
+    {
+        $this->initialize();
+        list($package, $path) = $this->getPackageAndPathByPublicPath($path);
+        return $this->getPublicPackageResourceUri($package, $path);
+    }
+
+    /**
+     * @param string $path
+     * @return array
+     * @throws Exception
+     * @api
+     */
+    public function getPackageAndPathByPublicPath($path)
+    {
+        $matches = array();
+        if (preg_match(self::PUBLIC_RESSOURCE_REGEXP, $path, $matches) === 1) {
+            $package = $matches[1];
+            $path = $matches[2];
+        } else {
+            throw new Exception(sprintf('The path "%s" which was given must point to a public resource.', $path), 1450358448);
+        }
+        return [$package, $path];
```

Comment:
could be shortened by !== 1 and then throw the exception and ``return [$matches[1], $matches[2]];``

could even make sense to use named groups in the regex to replace 1 and 2 with ``packageKey`` and ``relativePathAndFilename ``